File: 2020/Days/Day6-1.py
with open("Day6Input.txt") as f:
    lines = f.read().split("\n")
total, charachters= 0, []
for line in lines:
    if line == "":
        total += len(charachters)
        charachters = []
    for char in line:
        if char not in charachters:
            charachters.append(char)
print(total)

File: 2022/advent/src/bin/day6-1.rs
use std::collections::HashSet;

use advent::lib;
//use std::collections::HashSet;

fn main(){
    let lines: Vec<String> = lib::read_lines_vec("./in/day6input.txt");
    for line in lines{
        for p in 0..line.len()-3{
            let mut chars_set = HashSet::<char>::new();
            for c in line[p..p+4].to_string().chars(){
                chars_set.insert(c);
            }
            if chars_set.len() == 4{
                println!("{}", 4+p);
                break
            }
        }
    }
}